gpbft/gpbft.go

```diff
@@ -407,19 +429,24 @@ func (i *instance) validateMessage(msg *GMessage) error {
 	if msg.Vote.Instance != i.instanceID {
 		return xerrors.Errorf("message for wrong instance %d, expected %d", msg.Vote.Instance, i.instanceID)
 	}
+	// Ensure all participants are proposing the same instance data. This data is based on the
+	// _base_, so everyone should agree.
+	if !msg.Vote.Data.Eq(i.instanceData) {
+		return xerrors.Errorf("message for instance %d disagrees on the instance data", msg.Vote.Instance)
+	}
```

Comment:
When you rebase against #264, this will not be part of validation, but can only be checked when receiving a message.